components/core/src/os/process/windows_child.rs

```diff
@@ -776,37 +770,44 @@ impl RawHandle {
         }
     }
 
-    pub fn overlapped_result(&self, overlapped: *mut OVERLAPPED, wait: bool) -> io::Result<usize> {
-        unsafe {
-            let mut bytes = 0;
-            let wait = if wait { TRUE } else { FALSE };
-            let res =
-                cvt({ ioapiset::GetOverlappedResult(self.raw(), overlapped, &mut bytes, wait) });
-            match res {
-                Ok(_) => Ok(bytes as usize),
-                Err(e) => {
-                    if e.raw_os_error() == Some(ERROR_HANDLE_EOF as i32)
-                       || e.raw_os_error() == Some(ERROR_BROKEN_PIPE as i32)
-                    {
-                        Ok(0)
-                    } else {
-                        Err(e)
-                    }
+    pub unsafe fn overlapped_result(self,
+                                    overlapped: *mut OVERLAPPED,
+                                    wait: bool)
+                                    -> io::Result<usize> {
+        let mut bytes = 0;
+        let wait = if wait { TRUE } else { FALSE };
+        let res = cvt({ ioapiset::GetOverlappedResult(self.raw(), overlapped, &mut bytes, wait) });
+        match res {
+            Ok(_) => Ok(bytes as usize),
+            Err(e) => {
+                if e.raw_os_error() == Some(ERROR_HANDLE_EOF as i32)
+                   || e.raw_os_error() == Some(ERROR_BROKEN_PIPE as i32)
+                {
+                    Ok(0)
+                } else {
+                    Err(e)
                 }
             }
         }
     }
 
-    pub fn cancel_io(&self) -> io::Result<()> {
+    pub fn cancel_io(self) -> io::Result<()> {
         unsafe { cvt(ioapiset::CancelIo(self.raw())).map(|_| ()) }
     }
 
+    // This allow is on purpose. If you take clippy's suggestion and change
+    // pub fn read_to_end(&self, buf: &mut Vec<u8>)
+    // to
+    // pub fn read_to_end(self, buf: &mut Vec<u8>)
+    // then the function recurses endlessly. There's likely a better way to fix this that's more
+    // correct, but in the interest of time, right now I'm just going to allow it.
+    #[allow(clippy::trivially_copy_pass_by_ref)]
     pub fn read_to_end(&self, buf: &mut Vec<u8>) -> io::Result<usize> {
         let mut me = self;
         (&mut me).read_to_end(buf)
     }
```

Comment:
Shouldn't this pass in a `&mut self` instead?

It also seems like `RawHandle` wants to implement `std::io::Read`, since its `read` implementation is the same, essentially, once you look past `RawHandle` implementing `Copy`. That would also give us a `read_to_end` implementation automatically.

It seems like `RawHandle` might also want to implement `std::io::Write`, as well.

I'm also wondering if `RawHandle` _should_ implement `Copy`, given how it seems to be muddying the waters.

In light of this, it may be a better idea to back out these `trivially_copy_pass_by_ref` Clippy changes in favor of a more considered refactoring... with a healthy amount of testing in place first.

Reply:
Yeah, on second thought, I'm just going to allow all these clippy changes for now and we can always revisit later.  This entire module feels fraught with peril.

Reply:
yeah I think it is saved by this:
```
impl<'a> Read for &'a RawHandle {
    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> { (**self).read(buf) }

    fn read_to_end(&mut self, buf: &mut Vec<u8>) -> io::Result<usize> {
        unsafe { read_to_end_uninitialized(self, buf) }
    }
}
```
Alot of this code was literally copy/pasted from core rust which unfortunately limits the args you can pass to the underlying `CreateProcess` API that we needed finer control over. Looking back at  https://github.com/rust-lang/rust/blob/00856722bad5e9d96048319fb41f4b7e249820cd/src/libstd/sys/windows/handle.rs today these `read_to_end` calls have been refactored out. But back to the 2016 code it was copied and tweaked from, I did miss a `deref`. We should probably look to see how we can at least better align this with current day core and investigate if they have made any of this public or externally usable.

But definitely want to be cautious.

Reply:
Oh man, that's _evil_!

Reply:
^^^ is definitely outside of the scope of this PR

Reply:
/me runs screaming